src/ray/object_manager/object_manager_client_connection.h

```diff
@@ -7,63 +7,141 @@
 
 #include <boost/asio.hpp>
 #include <boost/asio/error.hpp>
+#include <boost/bind.hpp>
 #include <boost/enable_shared_from_this.hpp>
 
+#include "common/state/ray_config.h"
+#include "ray/common/client_connection.h"
 #include "ray/id.h"
 
 namespace ray {
 
-struct SendRequest {
-  ObjectID object_id;
-  ClientID client_id;
-  int64_t object_size;
-  uint8_t *data;
-};
-
-// TODO(hme): Document public API after integration with common connection.
-class SenderConnection : public boost::enable_shared_from_this<SenderConnection> {
+class SenderConnection : public ServerConnection<boost::asio::ip::tcp>,
+                         public boost::enable_shared_from_this<SenderConnection> {
  public:
+  /// Pointer type for this class.
   typedef boost::shared_ptr<SenderConnection> pointer;
```

Comment:
actually, in the header files, it's best to be explicit about types. I'd rather we just used `boost::shared_ptr<SenderConnection>` explicitly here instead of `pointer` (which is a very generic name that a reader would have to go and lookup the typedef for)